### PR TITLE
fix: not remove file by mistakes

### PR DESCRIPTION
**Pull request by chenminjian**
ref: #https://github.com/ipfs/go-ipfs/issues/6127

### CODE REVIEW

File: ops.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func Mv(r *Root, src, dst string) error {
 		return err
 	}
 
+	if srcDirObj == dstDir && srcFname == filename {
```
Comment (magik6k) [inline ops.go]: I don't think we should be comparing objects like that, looking at paths should be enough:
```suggestion
	if srcDir == dstDirStr && srcFname == filename {
```

**Comment by Stebalien** [comment]
That's a nasty bug, thanks for catching it! Mind adding a test to `test/sharness/t0250-files-api.sh`?

**Comment by chenminjian** [comment]
OK, I would like to do that.

**Comment by Stebalien** [comment]
Thanks!

**Pull request merged**